jaxws-ri/tools/wscompile/src/test/java/com/sun/tools/ws/test/processor/modeler/annotation/NillableArrayTest.java

```diff
@@ -0,0 +1,55 @@
+package com.sun.tools.ws.test.processor.modeler.annotation;
+
+import com.sun.tools.ws.wscompile.WsgenTool;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import javax.xml.parsers.DocumentBuilderFactory;
+import junit.framework.Assert;
+import junit.framework.TestCase;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+public class NillableArrayTest extends TestCase {
+    public void testNillableArray() {
+        File destDir;
+        List<String> options;
+
+        destDir = new File(System.getProperty("java.io.tmpdir"), NillableArrayTest.class.getSimpleName());
+        destDir.mkdirs();
+        options = new ArrayList<>();
+        options.add("-d");
+        options.add(destDir.getAbsolutePath());
+        options.add("-cp");
+        options.add(System.getProperty("java.class.path") + File.pathSeparator + System.getProperty("jdk.module.path"));
+        options.add("com.sun.tools.ws.test.processor.modeler.annotation.NillableTest");
+        options.add("-wsdl");
+
+        Properties props = System.getProperties();
+        props.setProperty("com.sun.xml.ws.jaxb.allowNonNillableArray","true");

```

Comment:
This should be reset after the test run so other tests running in the same VM are not affected. Or am I missing something?